src/realm/object-store/sync/app_utils.cpp

```diff
@@ -43,15 +41,38 @@ AppUtils::find_header(const std::string& key_name, const std::map<std::string, s
     return nullptr;
 }
 
+bool AppUtils::contains_string(const std::string& haystack, const std::string& needle)
```

Comment:
[nit] seems like this is an internal helper, can just make it a standalone function and remove it from the header

also the name does not make it clear how it's different from `.find()`, can you update the name to include something about case-insensitivity?

Reply:
I renamed the function, but have decided to keep it in _app\_utils.h_ since I plan on updating _app.cpp_ in the future with this helper in the places where we check for `application/json` in the response headers.